test/object-store/sync/sync_test_utils.cpp

```diff
@@ -476,13 +476,191 @@ struct BaasClientReset : public TestClientReset {
     TestAppSession& m_test_app_session;
 };
 
+struct BaasFLXClientReset : public TestClientReset {
+    BaasFLXClientReset(const Realm::Config& local_config, const Realm::Config& remote_config,
+                       const TestAppSession& test_app_session)
+        : TestClientReset(local_config, remote_config)
+        , m_test_app_session(test_app_session)
+    {
+        REALM_ASSERT(m_local_config.sync_config->flx_sync_requested);
+        REALM_ASSERT(m_remote_config.sync_config->flx_sync_requested);
+        REALM_ASSERT(m_local_config.schema->find(c_object_schema_name) != m_local_config.schema->end());
+    }
+
+    void wait_for_initial_sync()
+    {
+        // Reduce the liklihood of getting ProtocolError 229:
+        // "attempted to start a session while initial sync is in progress"
+        // This is not a fatal error, and sync should retry within 5 minutes,
```

Comment:
this should retry within 2 seconds and will exponentially back off up to 5 minutes. if you're still getting this error after 5 mins, I'd say something has gone wrong on the server side. there is some planned work to remove this polling behavior.

I think we should just remove this function. If we need to change the backoff behavior in the client so that we don't wait as much I'd be fine with that. I def picked the values for how we backoff here subjectively during FLX sync development.

Reply:
The client side backoff seems to be working. But I have been stuck in that loop for more than 5 minutes more than once and it was super annoying while testing this. Maybe my local baas docker instance was just overloaded with old app state and got into a weird state or something though.